craft_application/launchpad/models/recipe.py

```diff
@@ -326,11 +326,16 @@ def build(
         return self._build(deadline, request_build_kwargs)
 
 
-class CharmRecipe(_StoreRecipe):
-    """A recipe for a charm.
+class _StandardRecipe(_StoreRecipe):
+    """A base class for recipes with common launchpad builds."""
 
-    https://api.launchpad.net/devel.html#charm_recipe
-    """
+    ARTIFACT: ClassVar[str]
+    """The artifact type this recipe creates: charm, rock, etc."""
+
+    @classmethod
+    def _lp_recipe(cls, lp: Launchpad) -> Any:  # noqa: ANN401 (use of any)
```

Comment:
sure, that's fine. I don't think `_lp_recipe` would be confused with an attribute because it takes a parameter but I'm not against `_get_lp_recipe` either. Updated.